tool/src/demo_gen/terminus.rs

```diff
@@ -173,6 +173,16 @@ impl RenderTerminusContext<'_, '_> {
     /// That is, if there exists a string/buffer output. (Also called "returning a writeable")
     pub fn is_valid_terminus(method: &Method) -> bool {
         method.output.success_type().is_write()
+            || method
+                .output
+                .success_type()
+                .as_type()
+                .is_some_and(|t| matches!(t, hir::OutType::Enum(_) | hir::OutType::Primitive(_)))
+                && method
+                    .param_self
+                    .iter()
+                    .all(|s| !s.ty.is_mutably_borrowed())
+                && method.params.iter().all(|p| !p.ty.is_mutably_borrowed())
```

Comment:
Issue: It looks to me like these conditions should apply to methods that return strings as well as methods that return primitives, but currently they are only applying to methods that return primitives due to operator precedence

https://doc.rust-lang.org/reference/expressions.html#r-expr.precedence

Even if this is intentional, please use parentheses so I don't need to check the precedence table

Reply:
it's probably not going to make a difference (doesn't for ICU4X right now), as mutators don't usually return strings (but they might return status codes). will change this though